### PR TITLE
Docker development environment, replace Vagrant

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Very close to the travis environment

Includes useful .sh and .bat scripts for running various things

### CODE REVIEW

**Comment by davidtaylorhq** [comment]
@nhyty please can you give this a try on windows and see if it works - instructions in the README

**Comment by davidtaylorhq** [comment]
So... while docker supports windows, it requires Hyper-V support, which is only in Win 10 Enterprise/Pro/Education. 
https://docs.microsoft.com/en-gb/virtualization/hyper-v-on-windows/reference/hyper-v-requirements

**Comment by wjdp** [comment]
Yeep, see body of #695

**Comment by davidtaylorhq** [comment]
Ah yes. @nhyty how did using docker-machine go?

**Comment by nhyty** [comment]
Shall check when at home in about 30 mins

**Comment by nhyty** [comment]
David and I are working on testing this on windows

**Comment by wjdp** [comment]
@davidtaylorhq I've changed the nodejs version we use from v4.2.6 to v6.10.0. See the README changes in #735 for a quick way to install.

**Comment by davidtaylorhq** [comment]
Doh! I spent ages getting that old version of node working :P

**Comment by wjdp** [comment]
Oop, spotted you're using the travis container. Can just use nvm then I think.

**Comment by wjdp** [comment]
Sorry!!! Just happened to be the one I had installed.

File: run_dev.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 installdep="echo '----- Install htmltest' &&\
     	rsync -Pr /home/travis/to_copy/_bin /home/travis/newtheatre/history-project/ &&\
     	echo '----- NPM Install' &&\
-    	npm install && \
+    	npm install --no-bin-links && \
```
Comment (wjdp) [inline run_dev.sh]: Ah yes, a rather useful feature that Windows just does not has.

Comment (samozzy) [inline run_dev.sh]: Just use bash on Windows, right? 😉

File: README.md [hunk 2]
```diff
@@ -43,43 +43,33 @@ To get more verbose output from Jekyll, run `export JEKYLL_LOG_LEVEL=debug` befo
...
-- `gulp build`
-- `gulp server`
-
-The vagrant box has port 8000 mapped to 8000 on your local machine, so `http://127.0.0.1:8000` should still work.
-
 ## Docker
 
 [Docker](https://www.docker.com) is a cross platform software container platform. Following these instructions will allow you to build
-and view the site on your local machine in an environment as close to the production system as possible.
+and view the site on your local machine in an environment as close to the production system as possible. You ne
```
Comment (nhyty) [inline README.md]: This appears to be half a sentence @davidtaylorhq

Comment (davidtaylorhq) [inline README.md]: Indeed it does

**Comment by davidtaylorhq** [comment]
Note that this PR also removes Vagrant. The current Vagrantfile is out of date, and does not work. Docker should be a good replacement in almost all development environments.

**Pull request merged**